packages/quip-cli/src/lib/cli-api.ts

```diff
@@ -121,9 +121,18 @@ const cliAPI = async (configPath: string, site: string) => {
     return doAPICall;
 };
 
-const platformHost = (site: string) => {
-    const subdomainCount = site.split(".").length - 2;
-    return subdomainCount > 0 ? `platform-${site}` : `platform.${site}`;
+export const platformHost = (site: string) => {
+    const siteParts = site.split(".");
+    if (site === "staging.quip.com") {
+        return "platform-staging.quip.com";
+    }
+    if (siteParts.length > 2) {
+        if (siteParts.slice(-2).join(".") === "onquip.com") {
+            return `platform.${siteParts.slice(-3).join(".")}`;
+        }
+        return `platform.${siteParts.slice(-2).join(".")}`;
+    }
+    return `platform.${site}`;
```

Comment:
This logic is derived from https://github.com/quip/quip/blob/675ac6e717fe32e93388a60eb5054dd32e05b185/lib/vpc_migration/util.py#L166